processor/cloningfanoutconnector.go

```diff
@@ -33,6 +33,18 @@ import (
 // clones of data before fanning out, which ensures each consumer gets their
 // own copy of data and is free to modify it.
 
+// CreateMetricsCloningFanOutConnector is a placeholder function for now.
+// It supposed to create an old type connector or a new type connector based on type of provided metrics consumer.
+func CreateMetricsCloningFanOutConnector(baseMetricsConsumers []consumer.BaseMetricsConsumer) consumer.MetricsConsumer {
+	// TODO: CreateMetricsCloningFanOutConnector doesn't support new type of consumers
```

Comment:
Can we make this fails loudly if we attempt to use this function until this TODO is done? Otherwise this may become a hard to catch latent race bug if we forget to implement it. `panic` is fine if we cannot return an error. I know we don't yet have any components that implement the new V2 interfaces, so the `panic` will not be actually hit. It will just be a safeguard that we will discover the problem quickly if we forget about it.

Reply:
Good point. Added panic